src/utilities/scripts/windio.py

```diff
@@ -0,0 +1,256 @@
+    Converts a snake_case string to PascalCase string.
+    Removes spaces and replaces forward slashes with 'DividedBy'
+
+    Args:
+        snake_str (str): The snake_case string to convert
+
+    Returns:
+        str: The camelCase string
+    """
+    # snake_case to PascalCase
+    components = snake_str.split('_')
+    modified = ''.join(x.title() for x in components)
+
+    # remove spaces
+    modified = modified.replace(' ', '')
+
+    # replace forward slashes with 'DividedBy'
+    if '/' in modified:
+        modified = modified.replace('/', 'DividedBy')
+
+    return modified
+
+def modify_variable_name(snake_str: str) -> str:
+    # remove spaces
+    modified = snake_str.replace(' ', '')
+
+    # replace forward slashes with 'DividedBy'
+    if '/' in modified:
+        modified = modified.replace('/', '_divided_by_')
+
+    return modified
+
+
+def build_structs(s: Struct, struct_schema: Schema, definition_map: dict, struct_map: dict[str, Struct]) -> None:
+    """
+    Builds the struct based on the schema
+
+    Args:
+        s (Struct): The struct to build
+        struct_schema (Schema): The schema of the struct
+        definition_map (dict): The definitions in the schema
+        struct_map (dict[str, Struct]): The structs that have been built
+
+    Returns:
+        None
+    """
+    # If struct has been generated, return
+    if struct_schema.generated:
+        return
+    # Mark struct as generated
+    struct_schema.generated = True
+
+    # Check if the struct is already in the map and create test struct to compare
+    if s.name in struct_map:
+        print(f"Struct {s.name} already exists")
+        # create a test field
+        s_test = Struct(s.name, s.desc)
+        for field_name, field_schema in struct_schema.properties.items():
+            test_field = Field(
+                name=modify_name(field_name),
+                name_yaml=modify_variable_name(field_name),
+                type=field_schema.type,
+                desc=field_schema.description,
+            )
+            build_type(test_field, field_schema, definition_map, struct_map)
+            s_test.fields.append(test_field)
+
+        # Now check if the struct in the map has the same fields as the struct we are trying to build
+        same = True
+        for field in s_test.fields:
+            if field not in struct_map[s.name].fields:
+                # If the field is not in the struct in the map, the struct is not the same
+                same = False
+                break
+        if same:
+            # If the struct is the same, return
+            print(f"Struct {s.name} is the same, returning")
+            return
+
+        # If the struct is not the same, add a suffix to the name by keeping track of the number of structs with the same name
+        print(f"Struct {s.name} is not the same, adding suffix")
+        i = 1
+        while f"{s.name}_{i}" in struct_map:
+            i += 1
+        s.name = f"{s.name}_{i}"
```

Comment:
Is this logic necessary for our usecase?  Do we expect IEAOntology_schema to have duplicate struct names?  If not, then we should just error out if s.name already exists.  If this is an expected use case, then let's not print to screen on 105, 127, and 131.

Reply:
> Do we expect IEAOntology_schema to have duplicate struct names?

We unfortunately need to handle this case -- the schema contains duplicate struct names with different makeup.